bake/bake.go

```diff
@@ -233,7 +237,7 @@ func (c Config) target(name string, visited map[string]struct{}) (*Target, error
 		}
 	}
 	t.Inherits = nil
-	tt = merge(merge(defaultTarget(), t), tt)
+	tt = merge(merge(merge(defaultTarget(), t), tt), overrides[name])
```

Comment:
Note: nothing is setting `c.Target[name]` (aka `t` here), so I believe we could remove the defaultTarget() merge and do `merge(t, tt)` directly.